pipe/section/section_impls/tagging_transformer/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+            let mut output = pin!(output);
+
+            loop {
+                futures::select! {
+                    cmd = section_channel.recv().fuse() => {
+                        if let Command::Stop = cmd? {
+                            return Ok(())
+                        }
+                    }
+                    stream = input.next() => {
+                        let mut stream = match stream{
+                            Some(stream) => stream,
+                            None => Err("input stream closed")?
+                        };
+                        loop {
+                            futures::select! {
+                                msg = stream.next().fuse() => {
+                                    match msg? {
+                                        Some(Chunk::DataFrame(df)) => {
+                                            section_channel.log(format!("transform got dataframe chunk from {}:\n{}",
+                                                stream.origin(),
+                                                pretty_print(&*df))).await?;
+
+                                            // Convert to a RecordBatch from the arrow crate
+                                            let rb = df_to_recordbatch(df)?;
+                                            let old_schema = rb.schema();
+                                            let old_cols = rb.columns();
+
+                                            // Create a new schema from the old schema and push on a new field
+                                            let mut builder = SchemaBuilder::from(old_schema.fields());
+                                            builder.push(Field::new(self.column.clone(), ArrowDataType::Utf8, false));
+                                            let new_schema = builder.finish();
+
+                                            // Push a new value onto the old columns
+                                            let mut values = old_cols
+                                                .iter()
+                                                .map(std::borrow::ToOwned::to_owned)
+                                                .collect::<Vec<_>>();
+                                            let tag = Arc::new(StringArray::from(vec![self.text.clone()]));
+                                            values.push(tag);
+
+                                            // create a new arrow RecordBatch from the schema and columns
+                                            let new_rb = ArrowRecordBatch::try_new(
+                                                Arc::new(new_schema),
+                                                values,
+                                            )?;
+
+                                            // create a message from the RecordBatch
+                                            let new_msg = ArrowMsg::new("sitm", vec![Some(new_rb.into())], None);
```

Comment:
@dmzmk how would ack-ing a message work w/ a transformer section? Do we pass the ack fn through to the final destination?

Reply:
I didn't realize it wasn't handling the acks. How are those supposed to behave, is it acking the message was handled by the transformer or the destination? And if it's the destination will there only ever be one destination and if not which one acks the message?

Reply:
Here's an example of how the ack is used in a source: https://github.com/mycelial/mycelial/blob/main/pipe/section/section_impls/sqlite_connector/src/source.rs#L198-L203 

The state is stored locally in a sqlite db, and when the daemon restarts, it uses the state to not re-send data that's been received on the other side already. 

As far as multiple destinations, I feel like that might be a bigger project on its own and maybe we can punt that question until we decide we're going to allow a section to have multiple outputs. 

So I guess because of the way the ack is used right now, which is to tell the source if the data should be sent again or not, we should pass it through to the destination.

Reply:
Okay, thank you for the help! This is a big enough chunk of work that I'm going to merge this PR and make another that passes through the Ack's.

Reply:
transform section doesn't really consume original message, just alters it in some way. In this case we need to take ack from original message and pass it along with newly build one.